### ISSUE TITLE
New-M365DSCConnection inconsistency in the profile selection

### ISSUE THREAD

**Issue by William-Francillette** [microsoft/Microsoft365DSC#2869]
#### The DSC configuration that is used to reproduce the issue (as detailed as possible)
```powershell
# insert configuration here
```

#### Version of the DSC module that was used ('dev' if using current dev branch)

I'm experiencing some inconsistency with 
 New-M365DSCConnection -Workload 'MicrosoftGraph' `
            -InboundParameters $PSBoundParameters `
            -ProfileName 'beta'

Depending on how it is run the profileName is not selected correctly

For example if I run get-TargetResource on its own the resource returns the right value and profile
But If I run  Start-DscConfiguration -Wait -Verbose -Force -Path .\M365TenantConfig, test-targetresource calls get-targetresource which initiate a connection and return the v1.0 profile and thus not returning value only present in the beta version.

If I force the profile selection using select-MgProfile -name beta I have no issue
I'm not sure what to do as I know we have released a pr previously to remove select-mgprofile

I have experienced the issue with a resource I am developing  based on Get-MgDeviceManagementDeviceConfiguration

**Comment by NikCharlebois** [comment]
Once we upgrade to V2 of the Microsoft GraphPowerShell SDK, we won't have that problem anymore.
https://devblogs.microsoft.com/microsoft365dev/microsoft-graph-powershell-v2-is-now-in-public-preview-half-the-size-and-will-speed-up-your-automations/#:~:text=Speed%20up%20your%20automations%20%20%20Version%20of,MB%20%28-23.63%25%29%20%20%20651.31%20MB%20%28-23.89%25%29%20

However, since we don't have any ETA for when this will GA, we need to get to the bottom of this. I will investigate.

**Comment by NikCharlebois** [comment]
I have been spending several hours over the past few days trying to investigate this. As part of my investigation, I created dummy resources: AADAABeta and AADAAVersion1 which both call into the Get-MgDeviceManagementDeviceConfiguration cmdlet in their Get-TargetResource. The Beta version off course calls New-M365DSCConnection with the beta profile selected and vice-versa for the Version1 one. Both resource print the current profile value inside of their Get and Set methods. I used the following configuration, alternating between v1 and beta and got the expected outcome.

```
AADAAVersion1 Version1-1
        {
            Credential           = $Credscredential;
            Ensure = "Present"
            DisplayName =" v1-1"
        }
        AADAABeta Beta-1
        {
            Credential           = $Credscredential;
            Ensure = "Present"
            DisplayName =" beta1"
        }
        AADAAVersion1 Version1-2
        {
            Credential           = $Credscredential;
            Ensure = "Present"
            DisplayName =" v1-2"
        }
        AADAABeta Beta-2
        {
            Credential           = $Credscredential;
            Ensure = "Present"
            DisplayName =" beta2"
        }
        AADAABeta Beta-3
        {
            Credential           = $Credscredential;
            Ensure = "Present"
            DisplayName =" beta3"
        }
```

![image](https://user-images.githubusercontent.com/2547149/217295879-8f9f6378-16cd-42c8-961d-db8b5b54e16d.png)

**Comment by William-Francillette** [comment]
That's interesting - I've experienced the issue with the assignment cmdlet in test: the policy from Get-TargetResource returns the right values but in Test-TargetResource it returns the v1.0 version of the cmdlet which doesn't return the DeviceAndAppManagementAssignmentFilterType 

```
$assignmentsValues = Get-MgDeviceManagementDeviceConfigurationAssignment -DeviceConfigurationId $Id
        $assignmentResult = @()
        foreach ($assignmentEntry in $AssignmentsValues)
        {
            $assignmentValue = @{
                dataType = $assignmentEntry.Target.AdditionalProperties.'@odata.type'
                deviceAndAppManagementAssignmentFilterType = $assignmentEntry.Target.DeviceAndAppManagementAssignmentFilterType.ToString()
                deviceAndAppManagementAssignmentFilterId = $assignmentEntry.Target.DeviceAndAppManagementAssignmentFilterId
                groupId = $assignmentEntry.Target.AdditionalProperties.groupId
            }
            $assignmentResult += $assignmentValue
        }
        $results.Add('Assignments', $assignmentResult)
```

To see the difference I use a write-verbose trace after  `$CurrentValues = Get-TargetResource @PSBoundParameters` in Test-TargetResource

![image](https://user-images.githubusercontent.com/79221284/217351674-1375e9ba-6574-4576-a32a-b0da10fc66a5.png)

**Comment by menswearUK** [comment]
I've been looking at this as I have the same issues (#2942) - The problem seems to be that the beta profile doesn't get set the first time New-M365DSCConnection is run (which explains why my proposed fix worked, it ensured the connection had been run twice by the time we needed the beta profile).
After that it works fine and switches between 1.0 and beta with no issues. The problem seems to stem from the connect-m365Tenant command in MSCloudLoginAssistant module.

Is this something that could be best fixed in the MSCloudLoginAssistant Module?  Alternatively I've proposed catching and correcting on the New-M365DSCConnection function.

select-MgProfile does clear the error but, for me at least, also generates masses of verbose output I'd rather avoid (unless you run it from within New-M365DSCConnection)

**Comment by andikrueger** [comment]
Does it make any difference if other authentication methods are used? e.g. switching from app based to certificate based authentication?

**Comment by William-Francillette** [comment]
I only use certificate based auth
I haven't check credentials

**Comment by NikCharlebois** [comment]
Since we are so close from getting the Graph PowerShell SDK v2.0 to GA, the recommendation is to wait for this to fix itself. As a reminder, with v2.0 beta and v1.0 calls are decoupled into their own modules and therefore we aren't required to switch profiles.

**Comment by William-Francillette** [comment]
@andikrueger This was fixed by #2988 - it can be closed now
thanks
